conda_build/skeletons/cran.py

```diff
@@ -1165,6 +1166,17 @@ def skeletonize(in_packages, output_dir=".", output_suffix="", add_maintainer=No
         if 'R' not in dep_dict:
             dep_dict['R'] = ''
 
+        os_type = cran_package.get("OS_type", '')
+        if os_type != 'unix' and os_type != 'windows' and os_type != '':
+            print("Unknown OS_type: {} in CRAN package".format(os_type))
+            os_type = ''
+        if os_type == 'unix':
+            d['skip_os'] = 'skip: True  # [not unix]'
```

Comment:
I think you need to add ``d["noarch_generic"] = ""`` if any of ``OS_type`` is present
```suggestion
            d['skip_os'] = 'skip: True  # [not unix]'
            d["noarch_generic"] = ""
```

Reply:
I added testcase for it.  Overriding here the 'noarch: generic' is IMHO correct, but I guess @mingwandroid  has here a strong opinion